libpod/container_internal.go

```diff
@@ -818,7 +814,10 @@ func (c *Container) cleanupStorage() error {
 	c.state.Mountpoint = ""
 	c.state.Mounted = false
 
-	return c.save()
+	if c.valid {
```

Comment:
I wonder how this is not erroring, given that the container looks like it's still valid when we get here... Maybe the DB RemoveContainer calls also set it. Not a bad thing, really.